nixos/modules/services/x11/desktop-managers/gnome.nix

```diff
@@ -189,6 +189,12 @@ in
           '';
         };
       };
+
+      waylandOnly = lib.mkOption {
+        type = lib.types.bool;
+        default = false;
+        description = "Whether to exclusively use the Wayland session.";
```

Comment:
Maybe we should manage expectations that this will still keep X11 dependencies in most packages?